chrome/content/zotero/xpcom/zotero.js

```diff
@@ -381,7 +381,26 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 					return false;
 				}
 			}
-			
+			if(Zotero.File.isDropboxDirectory(dataDir.path) && Zotero.Prefs.get('warnDropboxDataDir')){
+				let check = {value: false};
+				ps = Services.prompt;
+				index = ps.confirmEx(null,
```

Comment:
Per Javascript spec, functions are actually preprocessed before running and all `var` declarations are moved to the front of the function, so even if you have variables declared with `var` inside conditionals which don't run, they are still available within the function scope. Valid concern though, I will turn off the IDE warnings for multiple declarations (as I assume there will is existing code with multiple var declarations) and go for `let` in this case, since we're moving towards ES6.